contrib/nydusify/cmd/nydusify.go

```diff
@@ -700,8 +713,40 @@ func main() {
 				if err != nil {
 					return err
 				} else if backendConfig == "" {
-					// TODO get auth from docker configuration file
-					return errors.Errorf("backend configuration is empty, please specify option '--backend-config'")
+
+					backendType = "registry"
```

Comment:
Can we share these codes with `check` subcommand implementation?

Reply:
I have checked this part, and find it will be bad to observe the logic of the code if it shares these codes with check subcommand implementation due to the different parameter. The code is pasted as follows
```
//contrib/nydusify/cmd/nydusify.go
if err != nil {
    return err
} else if backendConfig == "" {

    backendType = "registry"
    parsed, err := reference.ParseNormalizedNamed(c.String("target"))

    if err != nil {
        return err
    }

    host := reference.Domain(parsed)
    repo := reference.Path(parsed)

    config := dockerconfig.LoadDefaultConfigFile(os.Stderr)
    authConfig, err := config.GetAuthConfig(host)
    if err != nil {
        return errors.Wrap(err, "get docker registry auth config")
    }

    var auth string
    if authConfig.Username != "" && authConfig.Password != "" {
        auth = base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%s:%s", authConfig.Username, authConfig.Password)))
    }
    skipVerify := false
    if c.Bool("target-insecure") {
        skipVerify = true
    }
    scheme := "http"

    backendConfigStruct := rule.RegistryBackendConfig{scheme, host, repo, auth, skipVerify}
    bytes, err := json.Marshal(backendConfigStruct)
    if err != nil {
        return errors.Wrap(err, "parse registry backend config")
    }
    backendConfig = string(bytes)

}


//contrib/nydusify/pkg/checker/rule/filesystem.go
if rule.NydusdConfig.BackendType == "" {
    rule.NydusdConfig.BackendType = "registry"

    if rule.NydusdConfig.BackendConfig == "" {
        config := dockerconfig.LoadDefaultConfigFile(os.Stderr)
        authConfig, err := config.GetAuthConfig(host)
        if err != nil {
            return nil, errors.Wrap(err, "get docker registry auth config")
        }

        var auth string
        if authConfig.Username != "" && authConfig.Password != "" {
            auth = base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%s:%s", authConfig.Username, authConfig.Password)))
        }
        skipVerify := false
        if rule.TargetInsecure {
            skipVerify = true
        }
        scheme := "https"
        if rule.PlainHTTP {
            scheme = "http"
        }

        backendConfig := RegistryBackendConfig{scheme, host, repo, auth, skipVerify}
        bytes, err := json.Marshal(backendConfig)
        if err != nil {
            return nil, errors.Wrap(err, "parse registry backend config")
        }
        rule.NydusdConfig.BackendConfig = string(bytes)
    }
}
```

Reply:
I think we can create a func named `NewRegistryBackendConfig(...) (string, error)` to reuse the part of codes at least, right?